### PR TITLE
add publish to GPR support

### PR DESCRIPTION
**Pull request by o-az**
This adds support for publishing to GitHub Package Registry (https://github.com/huggingface/huggingface.js/issues/47).

I can merge the two `'*-publish.yml` workflows into one once `changesets` support is added 💫

### CODE REVIEW

File: .github/workflows/hub-publish.yml [hunk 1]
```diff
@@ -42,3 +42,39 @@ jobs:
...
+        run: |
+          echo "//npm.pkg.github.com:_authToken=${{ secrets.GITHUB_TOKEN }}" >> .npmrc
+
+      - name: 'Publish'
+        working-directory: packages/hub
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          pnpm publish --access public --registry https://npm.pkg.github.com --no-git-checks
```
Comment (coyotte508) [inline .github/workflows/hub-publish.yml:80]:
```suggestion
      # hack - reuse actions/setup-node@v3 just to set a new registry 
      - uses: actions/setup-node@v3
        with:
          node-version: "18"
          registry-url: "https://npm.pkg.github.com"
      - run: pnpm publish .
        env:
          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/inference-publish.yml [hunk 2]
```diff
@@ -42,3 +42,41 @@ jobs:
...
+        run: |
+          echo "//npm.pkg.github.com:_authToken=${{ secrets.GITHUB_TOKEN }}" >> .npmrc
+
+      - name: 'Publish'
+        working-directory: packages/inference
+        env:
+          NPM_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          pnpm publish --access public --registry https://npm.pkg.github.com --no-git-checks
```
Comment (coyotte508) [inline .github/workflows/inference-publish.yml:82]:
```suggestion
      # hack - reuse actions/setup-node@v3 just to set a new registry 
      - uses: actions/setup-node@v3
        with:
          node-version: "18"
          registry-url: "https://npm.pkg.github.com"
      - run: pnpm publish .
        env:
          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by coyotte508 — CHANGES_REQUESTED** [review]
I had several try/errors on this branch: https://github.com/huggingface/huggingface.js/tree/test-github-publish, so I hope my suggestions are correct.

Basically we should publish both to npm and github packages at once, so that the publishing to github happens after the version was bumped up (otherwise `publish-gpr` and `version_and_release` would run in parallel). 

And no need to specify `working-directory: packages/hub` on the `run` commands, it's already a global setting ealier in the file.

**Review by o-az — COMMENTED** [review]
accepted @coyotte508 change requests.

**Review by coyotte508 — APPROVED** [review]

**Comment by coyotte508** [comment]
Thanks!

**Pull request merged**